File: src/api/backup.ts
// api/backup.ts —— 备份/还原/调度命令

import { invoke } from './client';
import type {
  BackupRequest,
  BackupResult,
  RestoreRequest,
  RestoreResult,
  ScheduleRequest,
} from '../types/backup';

export const backupApi = {
  execute: (req: BackupRequest): Promise<BackupResult> =>
    invoke<BackupResult>('backup_execute', {
      profile: req.conn,
      options: req.options,
      outputPath: req.outputPath,
      schema: req.schema,
      selectedTables: req.selectedTables,
      selectedViews: req.selectedViews,
      selectedRoutines: req.selectedRoutines,
    }),

  restore: (req: RestoreRequest): Promise<RestoreResult> =>
    invoke<RestoreResult>('restore_execute', {
      profile: req.conn,
      request: req,
    }),

  scheduleAdd: (req: ScheduleRequest): Promise<number> =>
    invoke<number>('schedule_add', { request: req }),

  scheduleRemove: (scheduleId: number): Promise<boolean> =>
    invoke<boolean>('schedule_remove', { scheduleId }),

  scheduleList: (): Promise<[number, ScheduleRequest][]> =>
    invoke<[number, ScheduleRequest][]>('schedule_list'),
};

File: src-tauri/src/commands/backup.rs
use tauri::State;
use crate::errors::{AppResult, AppError};
use crate::models::backup::*;
use crate::core::backup_restore::scheduler::SchedulerHandle;
use crate::core::pool::manager::PoolRegistry;
use crate::models::connection::ConnectionProfile;

#[tauri::command]
pub async fn backup_execute(
    profile: ConnectionProfile,
    options: BackupOptions,
    output_path: String,
    schema: String,
    selected_tables: Option<Vec<String>>,
    selected_views: Option<Vec<String>>,
    selected_routines: Option<Vec<String>>,
    state: State<'_, PoolRegistry>,
) -> AppResult<BackupResult> {
    let pool_id = state.get_or_create_pool(&profile).await?;
    let (adapter, handle) = state.get_pool_info(pool_id).ok_or(AppError::PoolNotFound(pool_id))?;
    crate::core::backup_restore::backup::execute_backup(
        &*adapter, &handle, &options, &output_path, &schema, &profile,
        selected_tables.as_deref(), selected_views.as_deref(), selected_routines.as_deref(),
    ).await
}

#[tauri::command]
pub async fn restore_execute(profile: ConnectionProfile, request: RestoreRequest, state: State<'_, PoolRegistry>) -> AppResult<RestoreResult> {
    let pool_id = state.get_or_create_pool(&profile).await?;
    let (adapter, handle) = state.get_pool_info(pool_id).ok_or(AppError::PoolNotFound(pool_id))?;
    crate::core::backup_restore::restore::execute_restore(&*adapter, &handle, &request).await
}

#[tauri::command]
pub fn schedule_add(request: ScheduleRequest, scheduler: State<'_, SchedulerHandle>) -> AppResult<u64> {
    Ok(scheduler.add(request))
}

#[tauri::command]
pub fn schedule_remove(schedule_id: u64, scheduler: State<'_, SchedulerHandle>) -> AppResult<bool> {
    Ok(scheduler.remove(schedule_id))
}

#[tauri::command]
pub fn schedule_list(scheduler: State<'_, SchedulerHandle>) -> AppResult<Vec<(u64, ScheduleRequest)>> {
    Ok(scheduler.list())
}
